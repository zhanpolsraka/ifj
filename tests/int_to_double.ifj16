//0
class Main{
	static void run(){
		int a = 1;
		double b = a;
	}
}