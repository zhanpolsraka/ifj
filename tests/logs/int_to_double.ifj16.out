---------------FILENAME: tests/int_to_double.ifj16---------------


Instruction stack item [7] -> class with name "Main"
Instruction stack item [6] -> function with name "run"
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
ASSIGN 1 to 0x24cb4e0
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
ASSIGN 1 to 0x24cb5a0
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Function -> run
Lokal variables: 
	Variable -> a, type INTEGER, value [1]
	Variable -> b, type DOUBLE, value [1.000000]


--------------INSTRUCTION LIST-------------

Instruction stack item [7] -> class with name "Main"
Instruction stack item [6] -> function with name "run"
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [3] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



